tests/sentry/sentry_metrics/querying/data/test_api.py

```diff
@@ -1649,3 +1649,213 @@ def test_query_with_basic_formula_and_coefficient_operators(self):
             assert meta[0][1]["unit_family"] == expected_unit_family
             assert meta[0][1]["unit"] == expected_unit
             assert meta[0][1]["scaling_factor"] is None
+
+    @with_feature("organizations:ddm-metrics-api-unit-normalization")
+    def test_filter_project_mapping(self) -> None:
+        mql = self.mql("sum", TransactionMRI.DURATION.value, "project:bar")
+        query = MQLQuery(mql)
+
+        results = self.run_query(
+            mql_queries=[query],
+            start=self.now() - timedelta(minutes=30),
+            end=self.now() + timedelta(hours=1, minutes=30),
+            interval=3600,
+            organization=self.project.organization,
+            projects=[self.project],
+            environments=[],
+            referrer="metrics.data.api",
+        )
+        data = results["data"]
+        assert len(data) == 1
+        assert data[0][0]["by"] == {}
+        assert data[0][0]["series"] == [
+            None,
+            self.to_reference_unit(12.0),
+            self.to_reference_unit(9.0),
+        ]
+        assert data[0][0]["totals"] == self.to_reference_unit(21.0)
+
+    def setup_second_project(self):
+        self.new_project_1 = self.create_project(name="Bar Again")
+        for value, transaction, platform, env, time in (
+            (1, "/hello", "android", "prod", self.now()),
+            (3, "/hello", "android", "prod", self.now()),
+            (5, "/hello", "android", "prod", self.now()),
+            (2, "/hello", "android", "prod", self.now() + timedelta(hours=1, minutes=30)),
+            (5, "/hello", "android", "prod", self.now() + timedelta(hours=1, minutes=30)),
+            (8, "/hello", "android", "prod", self.now() + timedelta(hours=1, minutes=30)),
+        ):
+            self.store_metric(
+                self.new_project_1.organization.id,
+                self.new_project_1.id,
+                "distribution",
+                TransactionMRI.DURATION.value,
+                {
+                    "transaction": transaction,
+                    "platform": platform,
+                    "environment": env,
+                },
+                self.ts(time),
+                value,
+                UseCaseID.TRANSACTIONS,
+            )
+
+    def setup_third_project(self):
+        self.new_project_2 = self.create_project(name="Bar Yet Again")
```

Comment:
Sharing state in a test might backfire, since if we assume `self.new_project_2` exists in a test, we need to make sure either we call `setup_third_project` in that test or that we had at least one test run before that called that method. I would rather return the project from the function.